int8 val

int32 x
int32 y
int32 size

bool is_leaf
